rust/src/sip/sip.rs

```diff
@@ -147,6 +151,55 @@ impl SIPState {
         }
     }
 
+    fn parse_request_tcp(&mut self, flow: *const core::Flow, stream_slice: StreamSlice) -> AppLayerResult {
+        let input = stream_slice.as_slice();
+        if input.is_empty() {
+            return AppLayerResult::ok();
+        }
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            input.len() as i64,
+            SIPFrameType::Pdu as u8,
+        );
+        SCLogDebug!("ts: pdu {:?}", _pdu);
+
+        match sip_parse_request(input) {
+            Ok((_, request)) => {
+                sip_frames_ts(flow, &stream_slice, &request);
+                self.append_request(input, request);
+            }
+            Err(Err::Incomplete(_needed)) => {
+                let consumed = input.len();
+                let needed_estimation = input.len() + 1;
+                SCLogDebug!(
+                    "Needed: {:?}, estimated needed: {:?}",
+                    _needed,
+                    needed_estimation
+                );
+                self.set_event(SIPEvent::IncompleteData);
```

Comment:
For TCP I don't think this makes much sense. In theory, the data could trickle in one byte at a time but still be valid.

Reply:
Ok, let's remove it.